lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -67,7 +75,7 @@ def _build_kubeclient(api_version:, context:, endpoint_path: nil)
         ssl_options: kube_context.ssl_options,
         auth_options: kube_context.auth_options
       )
-      client.discover
+      client.discover if discover
```

Comment:
Discovery will crash against certain API endpoints, so we make it explicit so we can still use the client for those paths.